test/plugins/util/web.spec.js

```diff
@@ -222,10 +233,22 @@ describe('plugins/util/web', () => {
       res.end()
 
       expect(span.context().tags).to.have.property(RESOURCE_NAME, 'GET /foo/bar')
+      expect(span.context().tags).to.have.property(HTTP_ROUTE, '/foo/bar')
+    })
+
+    it('should not override user provided routes', () => {
+      req.method = 'GET'
+      span.setTag('http.route', '/custom')
+
+      web.enterRoute(req, '/foo')
+      res.end()
+
+      expect(span.context().tags).to.have.property(RESOURCE_NAME, 'GET /custom')
+      expect(span.context().tags).to.have.property(HTTP_ROUTE, '/custom')
```

Comment:
Should we add this assertion to the other web framework integrations?

It is mostly a "pass-through" tag value, in the sense that if the resource is being correctly populated today, then `http.route` probably will too.

Reply:
I think my question is, how do we know that when using `express` that the correct value is passed from `express` through the helper and set as `HTTP_ROUTE` tag?

Reply:
`express` would use `enterRoute()` and `exitRoute()` which uses a variable attached to the context. This value is only used if the `http.route` tag was not set by the user. If it was _not_ set by the user, then the tag is set by the value stored from the integration and then stored in the tag as well.